core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryStrategy.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.awscore.retry;
+
+import software.amazon.awssdk.annotations.SdkPublicApi;
+import software.amazon.awssdk.awscore.exception.AwsServiceException;
+import software.amazon.awssdk.awscore.internal.AwsErrorCode;
+import software.amazon.awssdk.core.internal.retry.SdkDefaultRetryStrategy;
+import software.amazon.awssdk.core.retry.RetryMode;
+import software.amazon.awssdk.retries.AdaptiveRetryStrategy;
+import software.amazon.awssdk.retries.DefaultRetryStrategy;
+import software.amazon.awssdk.retries.LegacyRetryStrategy;
+import software.amazon.awssdk.retries.StandardRetryStrategy;
+import software.amazon.awssdk.retries.api.RetryStrategy;
+
+/**
+ * Retry strategies used by clients when communicating with AWS services.
+ */
+@SdkPublicApi
+public class AwsRetryStrategy {
+
+    private AwsRetryStrategy() {
+    }
+
+    /**
+     * Retrieve the {@link SdkDefaultRetryStrategy#defaultRetryStrategy()} with AWS-specific conditions added.
+     */
+    public static RetryStrategy<?, ?> defaultRetryStrategy() {
+        return forRetryMode(RetryMode.defaultRetryMode());
+    }
+
+    /**
+     * Retrieve the appropriate retry strategy for the retry mode with AWS-specific conditions added.
+     */
+    public static RetryStrategy<?, ?> forRetryMode(RetryMode mode) {
+        switch (mode) {
+            case STANDARD:
+                return standardRetryStrategy();
+            case ADAPTIVE:
+                return adaptiveRetryStrategy();
+            case LEGACY:
+                return legacyRetryStrategy();
+            default:
+                throw new IllegalStateException("unknown retry mode: " + mode);
+        }
+    }
+
+    /**
```

Comment:
Since this is a public API ,
Can we have Java doc in this format

```java
    /**
     *  Description
     * @param strategy
     * @return
     */
```

Reply:
I was mostly mirroring the [code here](https://github.com/aws/aws-sdk-java-v2/blob/a90b989c77c1cbb7ac0e03b5561c07ef4ea83be7/core/aws-core/src/main/java/software/amazon/awssdk/awscore/retry/AwsRetryPolicy.java#L43) that does not follow this convention, but sure, I can add the `@param` and `@return` tags.